$OpenBSD: patch-src_corelib_global_qcompilerdetection_h,v 1.4 2017/12/27 16:27:06 rsadowski Exp $
1. Unbreak tests.
2. Make sure [[nodiscard]] is not used without explicit C++17 request.

Index: src/corelib/global/qcompilerdetection.h
--- src/corelib/global/qcompilerdetection.h.orig
+++ src/corelib/global/qcompilerdetection.h
@@ -756,7 +756,7 @@
 #      define Q_COMPILER_TEMPLATE_ALIAS
 #    endif
 #    if __has_feature(cxx_thread_local)
-#      if !defined(__FreeBSD__) /* FreeBSD clang fails on __cxa_thread_atexit */
+#      if !defined(__FreeBSD__) && !defined(__OpenBSD__) /* FreeBSD clang fails on __cxa_thread_atexit */
 #        define Q_COMPILER_THREAD_LOCAL
 #      endif
 #    endif
